tests/beta/tapes/test_tape.py

```diff
@@ -0,0 +1,711 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the QuantumTape"""
+import pytest
+import numpy as np
+
+import pennylane as qml
+from pennylane.beta.tapes import QuantumTape, NewCircuitGraph
+from pennylane.beta.queuing import expval, var, sample, probs, MeasurementProcess
+
+
+def TestOperationMonkeypatching():
+    """Test that operations are monkeypatched only within the quantum tape"""
+    with QuantumTape() as tape:
+        op = qml.RX(0.432, wires=0)
+        obs = qml.PauliX(wires="a")
+        measure = expval(qml.PauliX(wires="a"))
+
+    assert tape.operations == [op]
+    assert tape.observables == [obs]
+
+    # now create an old QNode
+    dev = qml.device("default.qubit", wires=[0, "a"])
+
+    @qml.qnode(dev)
+    def func(x):
+        global op
+        op = qml.RX(x, wires=0)
+        return qml.expval(qml.PauliX(wires="a"))
+
+    # this should evaluate without error
+    func(0.432)
+
+    assert func.circuit.operations == [op]
+
+
+class TestConstruction:
+    """Test for queuing and construction"""
+
+    @pytest.fixture
+    def make_tape(self):
+        ops = []
+        obs = []
+
+        with QuantumTape() as tape:
+            ops += [qml.RX(0.432, wires=0)]
+            ops += [qml.Rot(0.543, 0, 0.23, wires=0)]
+            ops += [qml.CNOT(wires=[0, "a"])]
+            ops += [qml.RX(0.133, wires=4)]
+            obs += [qml.PauliX(wires="a")]
+            expval(obs[0])
+            obs += [probs(wires=[0, "a"])]
```

Comment:
How come you are constructing an invalid tape here (since you measure the same wire three times)? Is that because it is never checked by the tape class itself, and convenient?

Reply:
I have to admit, I haven't added a validation here regarding measuring wires more than once... I was thinking we could come up with a better solution down the line